scubagoggles/rego/Drive.rego

```diff
@@ -0,0 +1,1588 @@
+    "ReportDetails": utils.ReportDetails(NonCompliantOUs4_1, NonCompliantGroups4_1),
+    "ActualValue": {"NonCompliantOUs": NonCompliantOUs4_1, "NonCompliantGroups": NonCompliantGroups4_1},
+    "RequirementMet": Status,
+    "NoSuchEvent": false
+}
+if {
+    Check4_1_OK
+    Conditions := {count(NonCompliantOUs4_1) == 0, count(NonCompliantGroups4_1) == 0}
+    Status := (false in Conditions) == false
+}
+
+#--
+
+###################
+# GWS.DRIVEDOCS.5 #
+###################
+
+#
+# Baseline GWS.DRIVEDOCS.5.1
+#--
+
+DriveId5_1 := utils.PolicyIdWithSuffix("GWS.DRIVEDOCS.5.1")
+
+LogMessage5_1 := "ENABLE_DOCS_ADD_ONS"
+
+Check5_1_OK if {
+    # not PolicyApiInUse
+    events := utils.FilterEventsOU(LogEvents, LogMessage5_1, utils.TopLevelOU)
+    count(events) > 0
+}
+
+Check5_1_OK if {PolicyApiInUse}
+
+NonComplianceMessage5_1 := "Users can install Google Docs add-ons from add-ons store."
+
+NonCompliantOUs5_1 contains {
+    "Name": OU,
+    "Value": NonComplianceMessage5_1
+}
+if {
+    # not PolicyApiInUse
+    some OU in utils.OUsWithEvents
+    Events := utils.FilterEventsOU(LogEvents, LogMessage5_1, OU)
+    count(Events) > 0
+    LastEvent := utils.GetLastEvent(Events)
+    LastEvent.NewValue != "false"
+    LastEvent.NewValue != "INHERIT_FROM_PARENT"
+}
+
+NonCompliantGroups5_1 contains {
+    "Name": Group,
+    "Value": NonComplianceMessage5_1
+}
+if {
+    # not PolicyApiInUse
+    some Group in utils.GroupsWithEvents
+    Events := utils.FilterEventsGroup(LogEvents, LogMessage5_1, Group)
+    count(Events) > 0
+    LastEvent := utils.GetLastEvent(Events)
+    LastEvent.NewValue != "false"
+    LastEvent.NewValue != "INHERIT_FROM_PARENT"
+}
+
+# NOT yet implemented in policy API
+#NonCompliantOUs5_1 contains {
+#    "Name": OU,
+#    "Value": NonComplianceMessage5_1
+#}
+#if {
+#    some OU, settings in input.policies
+#    DriveEnabled(OU)
+#    addOns := settings.drive_and_docs_drive_sdk.enableDriveSdkApiAccess
+#    addOns != false
+#}
+
+tests contains {
+    "PolicyId": DriveId5_1,
+    "Criticality": "Shall",
+    "ReportDetails": utils.NoSuchEventDetails(DefaultSafe, utils.TopLevelOU),
+    "ActualValue": "No relevant event for the top-level OU in the current logs",
+    "RequirementMet": DefaultSafe,
+    "NoSuchEvent": true
+}
+if {
+    # not PolicyApiInUse
+    DefaultSafe := false
+    not Check5_1_OK
+}
+
+tests contains {
+    "PolicyId": DriveId5_1,
+    "Criticality": "Shall",
+    "ReportDetails": utils.ReportDetails(NonCompliantOUs5_1, NonCompliantGroups5_1),
+    "ActualValue": {"NonCompliantOUs": NonCompliantOUs5_1,
+        "NonCompliantGroups": NonCompliantGroups5_1},
+    "RequirementMet": Status,
+    "NoSuchEvent": false
+}
+if {
+    Check5_1_OK
+    Conditions := {count(NonCompliantOUs5_1) == 0, count(NonCompliantGroups5_1) == 0}
+    Status := (false in Conditions) == false
+}
+#--
+
+###################
+# GWS.DRIVEDOCS.6 #
+###################
+
+#
+# Baseline GWS.DRIVEDOCS.6.1
+#--
```

Comment:
The report fails you no matter what you select for the drive for desktop settings

Reply:
![image](https://github.com/user-attachments/assets/45ba055c-d181-410d-acd3-f160a5ed2dab)

Reply:
The org we use for testing uses Google Workspace Enterprise Plus, so that isn't the issue.

Reply:
After implementing Google's policy "reduction" process, this is working for me
when I change settings in the Admin UI.